Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Collections/GridObjectCollection.cs

```diff
@@ -418,18 +420,48 @@ protected virtual void OnDrawGizmosSelected()
                     break;
             }
         }
-    
 
-        public void OnValidate()
+#region asset version migration
+#if UNITY_EDITOR
+
+        static GridObjectCollection()
+        {
+            // Patch on reload. 
+            // Need to defer patching as accessing the scene is not allowed in the static constructor.
+            EditorApplication.update += PatchAllInstancesDeferred;
```

Comment:
I'm surprised OnValidate did not work. Did you also try ExecuteInEditMode? 
https://docs.unity3d.com/ScriptReference/ExecuteInEditMode.html

It's great you unsubscribe from update once done...this all will still execute though every code change and recompile though. And FindObjectsOfType isn't cheap, especially dependent on the scene. This looks like it should work but i'm slightly worried about setting the precedent of adding this kind of editor overhead as we get more components that need to be migrated. Especially since in general we only need to perform the migration once per scene and then never again, so we searching/checking constantly for only one update (generally) in the lifetime of this component

I'm not exactly sure of a better solution yet. The benefit to mind of ExecuteInEditMode for like Awake() is that you don't need to findobjectsoftype and you can clean the code up a bit

Reply:
I was also surprised OnValidate did not work. The asset did not get marked as dirty / to save when I used OnValidate.

I didn't notice any slowdown when checking on every asset save but can see that having such a precedent for all assets could cause a slowdown.

ExecuteAlways + Awake is working, so going with that approach